src/wrapper_array.jl

```diff
@@ -176,3 +176,43 @@ end
 @propagate_inbounds function Base.setindex!(a::NamedDimsArray, value, inds...)
     return setindex!(parent(a), value, inds...)
 end
+
+############################################
+# show
+
+# 2-arg show, mainly for repr():
+function Base.show(io::IO, A::NamedDimsArray{L,T,N}) where {L,T,N}
+    print(io, "NamedDimsArray(")
+    show(io, parent(A))
+    print(io, ", ", N == 1 ? QuoteNode(L[1]) : L, ")")
+end
+
+# called by summary(), for main REPL printing:
+function Base.showarg(io::IO, A::NamedDimsArray{L,T,N}, outer) where {L,T,N}
+    print(io, "NamedDimsArray(")
+    Base.showarg(io, parent(A), false)
+    print(io, ", ", N == 1 ? QuoteNode(L[1]) : L, ")")
+end
+
+function Base.print_matrix(io::IO, A::NamedDimsArray)
+    s1 = string("↓ :", dimnames(A,1)) * "  "
+    if ndims(A)==2
+        s2 = string(" "^length(s1), "→ :", dimnames(A,2), "\n")
```

Comment:
Note that this gets the spacing wrong for some unicode, e.g. `NamedDimsArray(rand(3,4),(:x̂, :ŷ))` where `length("x̂")==2`.

Reply:
Actually, the problem is in Base, not here:
```
julia> Base.print_matrix(stdout, ones(3,4), " x: "); println()
 x: 1.0  1.0  1.0  1.0
    1.0  1.0  1.0  1.0
    1.0  1.0  1.0  1.0

julia> Base.print_matrix(stdout, ones(3,4), " x̂: "); println()
 x̂: 1.0  1.0  1.0  1.0
     1.0  1.0  1.0  1.0
     1.0  1.0  1.0  1.0
```

Reply:
is there an issue open to fix this in Base?